### DISCUSSION TITLE
Any instructions on how to build from source in Visual Studio?

### DISCUSSION THREAD

**Discussion by apn-hroussel** [ikvmnet/ikvm#413] [General]
Hello.

I have VS 2022 installed and was trying to build from source, to help me debug an issue I have with a converted jar.

I got the following error when trying to compile IKVM.Runtime (which I assume is the first one to compile?),

3>javac: file not found: C:\apps\ikvm-8.6.3\openjdk\build\linux-x86_64-normal-server-release\jdk\gensrc\sun\util\cldr\CLDRLocaleDataMetaInfo.java

which leads me to think I need to get the source for openjdk.

So I found a jdku92 at the following address on Github,

https://github.com/ikvmnet/jdk8u/tree/ikvm-jdk8u92-b34

But it doesn't contain a build directory.

Any help appreciated!

**Comment by wasabii** [comment]
The CONTRIBUTING.md file should talk about the various requirements, etc. Otherwise, you just open it in VS and build the various parts.
